### PR TITLE
Update Task Heartbeat to include Job_id

### PR DESCRIPTION
**Pull request by nharper285**
I made some quick tweaks to include job_id in the task_heartbeat logging.

### CODE REVIEW

File: src/api-service/__app__/queue_task_heartbeat/__init__.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def main(msg: func.QueueMessage, dashboard: func.Out[str]) -> None:
             task.heartbeat = datetime.utcnow()
             task.save()
     except ValidationError:
-        logging.error("invalid task heartbat: %s", raw)
+        logging.error("invalid task heartbeat: %s", raw)
```
Comment (bmc-msft) [inline src/api-service/__app__/queue_task_heartbeat/__init__.py:33]: 👍

File: src/pytypes/onefuzztypes/models.py [hunk 2]
```diff
@@ -516,6 +516,7 @@ class Job(BaseModel):
 
 class TaskHeartbeatEntry(BaseModel):
     task_id: UUID
+    job_id: UUID
```
Comment (bmc-msft) [inline src/pytypes/onefuzztypes/models.py:519]: This needs to be optional, to allow parsing heartbeat messages from in-progress tasks running started prior to an upgrade.
```suggestion
    job_id: Optional[UUID]
```

File: src/agent/onefuzz-agent/src/tasks/heartbeat.rs [hunk 3]
```diff
@@ -32,12 +32,13 @@ pub struct TaskContext {
 
 pub type TaskHeartbeatClient = HeartbeatClient<TaskContext, HeartbeatData>;
 
-pub async fn init_task_heartbeat(queue_url: Url, task_id: Uuid) -> Result<TaskHeartbeatClient> {
+pub async fn init_task_heartbeat(queue_url: Url, task_id: Uuid, job_id: Uuid) -> Result<TaskHeartbeatClient> {
```
Comment (bmc-msft) [inline src/agent/onefuzz-agent/src/tasks/heartbeat.rs:35]: Please run `cargo fmt` on the code.  (This suggestion implements the required change)

```suggestion
pub async fn init_task_heartbeat(
    queue_url: Url,
    task_id: Uuid,
    job_id: Uuid,
) -> Result<TaskHeartbeatClient> {
```

**Review by bmc-msft — CHANGES_REQUESTED** [review]

**Review by chkeita — APPROVED** [review]

**Review by bmc-msft — APPROVED** [review]

**Comment by bmc-msft** [comment]
Running integration tests

**Pull request merged**